dev/Common/Logging.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <Windows.h>
+#include <source_location>
+#include <sstream>
+
+namespace Common::Logging
+{
+    // use std::format in msg parameter to have a printf like varargs syntax for the debug message
+    inline void DebugLog(const wchar_t* msg,  const std::source_location location = std::source_location::current())
+    {
+        std::wstringstream filename ;
+        filename << location.file_name(); // converting char to wstring (and wchar_t implicitly)
+        // format is "foo.cpp (66) : debug message"
+        const std::wstring printmsg = std::format(L"{} ({}) : {}\n", filename.str(), location.line(), msg);
```

Comment:
it doesn't work because location_filen_name() returns a wstring. std::format doesn't support mixing string and wstring together and results in a compilation error

Reply:
thanks Scott